cmd/loopd/daemon.go

```diff
@@ -7,34 +7,71 @@ import (
 	"net/http"
 	"os"
 	"os/signal"
+	"path/filepath"
 	"runtime/pprof"
 	"sync"
 	"time"
 
 	proxy "github.com/grpc-ecosystem/grpc-gateway/runtime"
 	"github.com/lightninglabs/loop"
+	"github.com/lightninglabs/loop/lndclient"
 	"github.com/lightninglabs/loop/looprpc"
 	"google.golang.org/grpc"
 )
 
 // daemon runs loopd in daemon mode. It will listen for grpc connections,
 // execute commands and pass back swap status information.
 func daemon(config *config) error {
+	// Based on the network, we may need to swap our items like the
+	// swapserver or the default location of macaroons if a path isn't set.
+	switch config.Network {
+	case "testnet":
+		// If the user is targeting the testnet network, then we'll
+		// point towards the testnet swap server rather than the
+		// mainnet endpoint.
+		config.SwapServer = testnetServer
+
+		if config.Lnd.MacaroonPath == "" {
+			config.Lnd.MacaroonPath = filepath.Join(
+				lndclient.DefaultLndDir,
+				lndclient.DefaultDataDir,
+				lndclient.DefaultChainSubDir, "bitcoin",
+				"testnet",
+			)
+		}
+
+	case "mainnet":
+		if config.Lnd.MacaroonPath == "" {
+			config.Lnd.MacaroonPath = filepath.Join(
+				lndclient.DefaultLndDir,
+				lndclient.DefaultDataDir,
+				lndclient.DefaultChainSubDir, "bitcoin",
+				"mainnet",
+			)
+
+		}
+
+	case "simnet":
+		if config.Lnd.MacaroonPath == "" {
+			config.Lnd.MacaroonPath = filepath.Join(
+				lndclient.DefaultLndDir,
+				lndclient.DefaultDataDir,
+				lndclient.DefaultChainSubDir, "bitcoin",
+				"simnet",
+			)
+		}
+	}
+
```

Comment:
This default dir logic isn't applied when the `view` command is executed.

Reply:
And also not when starting the server. Probably better to move this to a location where it is reusable.

Reply:
Moved it to reside inside of `lndclient.NewLndServices`. As a result of this change, the move to make some default dir params public is no longer needed, so I removed it.